main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala

```diff
@@ -200,7 +200,13 @@ object Unification {
     */
   def expectTypeM(expected: Type, actual: Type, loc: SourceLocation)(implicit flix: Flix): InferMonad[Type] = {
     def handler(e: TypeError): TypeError = e match {
-      case _: TypeError.MismatchedTypes => TypeError.UnexpectedType(expected, actual, loc)
+      case _: TypeError.MismatchedTypes => 
+        (expected, actual) match {
```

Comment:
When looking at type constructors, it is more idiomatic to do:

`expected.typeConstructor, actual.typeConstructor match { case (Some(TypeCon.Native...)), ...) => `

Reply:
The reason to use `.typeConstructor` is that it allows us more easily to change the structure of types.